Doc/c-api/module.rst

```diff
@@ -249,52 +255,22 @@ or request "multi-phase initialization" by returning the definition struct itsel
       .. versionchanged:: 3.9
          No longer called before the module state is allocated.
 
-Single-phase initialization
-...........................
-
-The module initialization function may create and return the module object
-directly. This is referred to as "single-phase initialization", and uses one
-of the following two module creation functions:
-
-.. c:function:: PyObject* PyModule_Create(PyModuleDef *def)
-
-   Create a new module object, given the definition in *def*.  This behaves
-   like :c:func:`PyModule_Create2` with *module_api_version* set to
-   :c:macro:`PYTHON_API_VERSION`.
-
-
-.. c:function:: PyObject* PyModule_Create2(PyModuleDef *def, int module_api_version)
-
-   Create a new module object, given the definition in *def*, assuming the
-   API version *module_api_version*.  If that version does not match the version
-   of the running interpreter, a :exc:`RuntimeWarning` is emitted.
-
-   Return ``NULL`` with an exception set on error.
-
-   .. note::
-
-      Most uses of this function should be using :c:func:`PyModule_Create`
-      instead; only use this if you are sure you need it.
-
-Before it is returned from in the initialization function, the resulting module
-object is typically populated using functions like :c:func:`PyModule_AddObjectRef`.
-
 .. _multi-phase-initialization:
 
 Multi-phase initialization
 ..........................
 
-An alternate way to specify extensions is to request "multi-phase initialization".
+The preferred method to specify extensions is to request "multi-phase initialization".
 Extension modules created this way behave more like Python modules: the
 initialization is split between the *creation phase*, when the module object
 is created, and the *execution phase*, when it is populated.
 The distinction is similar to the :py:meth:`!__new__` and :py:meth:`!__init__` methods
 of classes.
 
-Unlike modules created using single-phase initialization, these modules are not
-singletons: if the *sys.modules* entry is removed and the module is re-imported,
-a new module object is created, and the old module is subject to normal garbage
-collection -- as with Python modules.
+Unlike modules created using the legacy single-phase initialization mechanism,
+these modules are not singletons: if the *sys.modules* entry is removed and
+the module is re-imported, a new module object is created, and the old module
+is subject to normal garbage collection -- as with Python modules.
```

Comment:
I think this shouldn't be a comparison. If we're calling single-phase “legacy” (which finally [seems to be the cunsensus](https://discuss.python.org/t/89262/46)), then this section should describe what modules *are*, and the single-phase section should describe the eldritch weirdness of that method.
Should I try to reword it that way? (I'd like to avoid merge conflicts.)

Reply:
> Should I try to reword it that way? (I'd like to avoid merge conflicts.)

@encukou feel free to push to this branch.

Reply:
> Should I try to reword it that way? (I'd like to avoid merge conflicts.)

@encukou feel free to push to this branch.